rs-matter/src/data_model/sdm/general_commissioning.rs

```diff
@@ -171,11 +153,11 @@ impl<'a> GenCommCluster<'a> {
                     Attributes::BreadCrumb(codec) => codec.encode(writer, 0),
                     // TODO: Arch-Specific
                     Attributes::RegConfig(codec) => {
-                        codec.encode(writer, RegLocationType::IndoorOutdoor as _)
+                        codec.encode(writer, RegulatoryLocationTypeEnum::IndoorOutdoor as _)
```

Comment:
I'm wondering: do we expect name clashes between the `struct` namespace and the `enum` namespace? Because if not (or if we can somehow avoid it), we can probably get rid of the `Enum` suffix? Or is it defined as `RegulatoryLocationTypeEnum` in the IDL itself?

Reply:
It is named like that in the IDL: https://github.com/project-chip/connectedhomeip/blob/master/src/controller/data_model/controller-clusters.matter#L1303  and also in the spec it is `11.10.4.2. RegulatoryLocationTypeEnum Type`

The suffixes indeed seem annoying, however I think for consistency we should probably accept them.